libs/utils/android/workload.py

```diff
@@ -88,6 +88,29 @@ def run(self, out_dir, collect='',
             **kwargs):
         raise RuntimeError('Not implemeted')
 
+    def setupTargetScript(self, actions):
+        """
+        Push a script to the target
+
+        :param actions: List of actions(commands) to run
+        :type actions: list(str)
+        """
+
+        actions = ['#!{} sh'.format(self._target.busybox)] + actions
+        actions = str.join('\n', actions)
+
+        self._scriptpath = '{}/remote_script.sh'.format(self._target.executables_directory)
+
+        self._target.execute('echo \"{}\" > {}'.format(actions, self._scriptpath))
+        self._target.execute('chmod +x {}'.format(self._scriptpath))
+
+    def runTargetScript(self):
+        """
+        Run the previously pushed script
+        """
+        if self._scriptpath:
+            self._target.execute(self._scriptpath)
```

Comment:
Indeed, it makes sense to encapsulate in `DummyTarget` all the functions required to setup and run the script.